compiler/src/parsing/ast_helper.re

```diff
@@ -319,6 +319,31 @@ module Expression = {
     mk(~loc, ~core_loc, ~attributes?, PExpLambda(a, b));
   let apply = (~loc, ~core_loc, ~attributes=?, a, b) =>
     mk(~loc, ~core_loc, ~attributes?, PExpApp(a, b));
+  let partial_apply = (~loc, ~core_loc, ~attributes=?, a, b) =>
+    mk(~loc, ~core_loc, ~attributes?, PExpPartial(a, b));
+  let total_apply_args = (~loc, ~core_loc, ~attributes=?, a, b) => {
+    let args =
+      List.map(
+        arg =>
+          {
+            paa_loc: arg.ppaa_loc,
+            paa_expr:
+              switch (arg.ppaa_expr) {
+              | ArgumentGiven(expr) => expr
+              | ArgumentHole(_) =>
+                raise(
+                  SyntaxError(
+                    loc,
+                    "To use partial application, prefix the function call with `partial`.",
```

Comment:
This was kind of a bandaid fix around me being a noob with LR(1) parsers and not being able to figure out how to create separate rules for partial and total applications without avoiding reduce/reduce conflicts. If anyone has insight on this could be done please let me know